pkg/apis/apps/v1alpha1/servicebindingrequest_types.go

```diff
@@ -21,76 +21,40 @@ type ServiceBindingRequestSpec struct {
 	EnvVarPrefix string `json:"envVarPrefix,omitempty"`
 
 	// BackingServiceSelector is used to identify the backing service operator.
-	//
-	// Refer: https://12factor.net/backing-services
-	// A backing service is any service the app consumes over the network as
-	// part of its normal operation. Examples include datastores (such as
-	// MySQL or CouchDB), messaging/queueing systems (such as RabbitMQ or
-	// Beanstalkd), SMTP services for outbound email (such as Postfix), and
-	// caching systems (such as Memcached).
-	//
-	// Example 1:
-	//	backingServiceSelector:
-	//		resourceKind: databases.example.org
-	//      resourceRef: mysql-database
-	// Example 2:
-	//	backingServiceSelector:
-	//		resourceKind: databases.example.org
-	//		resourceVersion: v1alpha1
-	//      resourceRef: mysql-database
 	BackingServiceSelector BackingServiceSelector `json:"backingServiceSelector"`
 
 	// ApplicationSelector is used to identify the application connecting to the
 	// backing service operator.
-	// Example 1:
-	//	applicationSelector:
-	//		matchLabels:
-	//			connects-to: postgres
-	//			environment: stage
-	//		resourceKind: Deployment
-	// Example 2:
-	//	applicationSelector:
-	//		matchLabels:
-	//			connects-to: postgres
-	//			environment: stage
 	ApplicationSelector ApplicationSelector `json:"applicationSelector"`
 }
 
+// ServiceBindingRequestStatus defines the observed state of ServiceBindingRequest
+// +k8s:openapi-gen=true
+type ServiceBindingRequestStatus struct {
+	// BindingStatus is the status of the service binding request.
+	BindingStatus string `json:"bindingStatus,omitempty"`
+	// Secret is the name of the intermediate secret
+	Secret string `json:"secret,omitempty"`
+	// ApplicationObjects contains all the application objects filtered by label
+	ApplicationObjects []string `json:"applicationObjects,omitempty"`
+}
+
 // BackingServiceSelector defines the selector based on resource name, version, and resource kind
 // +k8s:openapi-gen=true
 type BackingServiceSelector struct {
-	Group       *string `json:"group,omitempty"`
-	Version     string  `json:"version"`
-	Kind        string  `json:"kind"`
-	ResourceRef string  `json:"resourceRef"`
+	Group       string `json:"group,omitempty"`
+	Version     string `json:"version"`
+	Kind        string `json:"kind"`
+	ResourceRef string `json:"resourceRef"`
 }
 
-// ApplicationSelector defines the selector based on labels and resource kind
+// ApplicationSelector defines the selector based on labels and GVR
 // +k8s:openapi-gen=true
 type ApplicationSelector struct {
 	MatchLabels map[string]string `json:"matchLabels"`
-	Group       *string           `json:"group,omitempty"`
+	Group       string            `json:"group,omitempty"`
```

Comment:
Why did we remove the pointer? :) It's typically a good way to say if a specific field is optional hence nulllable?
Or, did you make `Group` mandatory in this? I guess the latter.. ah Yes.
( sorry thought out aloud :) )

Reply:
@sbose78 if the field is nullable or empty, should we then use `omitempty`? I thought the pointer reference over there were a typo, since no other fields were doing the same, and examples using Operator-SDK does not show that in a String.

Reply:
The pointer reference is for use in the code - During deserialization if the `Group` in the yaml is empty, the ` Group` in the struct would be nil if we specify it as a pointer  :) 
That along with the `omitempty` is to tell `operator-sdk generate openapi` that the specific field is to be marked as non-required in the openapi validation spec.